astroplan/core.py

```diff
@@ -162,171 +207,851 @@ def altaz(self, time, target=None, obswl=None):
 
         Parameters
         ----------
-        time : `~astropy.time.Time`
+        time : `~astropy.time.Time` or other (see below)
+            This will be passed in as the first argument to
+            the `~astropy.time.Time` initializer, so it can be anything that
+            `~astropy.time.Time` will accept (including a `~astropy.time.Time`
+            object).
+
+        Keywords: str, optional
+            previous
+            next
         """
         raise NotImplementedError()
 
-    def morning_nautical(self, time):
+    def moon_illumination(self, time):
         """
-        Returns the local morning (nautical) twilight time.
+        Returns a float giving the percent illumation.
 
         Parameters
         ----------
-        time : `~astropy.time.Time`
+        time : `~astropy.time.Time` or other (see below)
+            This will be passed in as the first argument to
+            the `~astropy.time.Time` initializer, so it can be anything that
+            `~astropy.time.Time` will accept (including a `~astropy.time.Time`
+            object).
         """
         raise NotImplementedError()
 
-    def morning_civil(self, time):
+    def moon_position(self, time):
         """
-        Returns the local morning (civil) twilight time.
+        Returns the position of the moon in alt/az.
 
         Parameters
         ----------
-        time : `~astropy.time.Time`
+        time : `~astropy.time.Time` or other (see below)
+            This will be passed in as the first argument to
+            the `~astropy.time.Time` initializer, so it can be anything that
+            `~astropy.time.Time` will accept (including a `~astropy.time.Time`
+            object).
         """
         raise NotImplementedError()
 
+    @u.quantity_input(horizon=u.deg)
+    def can_see(self, time, target, horizon=0*u.degree, ret_altaz=False):
+        '''
+        Is ``target`` above ``horizon`` at this ``time``?
+
+        Parameters
+        ----------
+        time : `~astropy.time.Time` or other (see below)
+            Time of observation. This will be passed in as the first argument to
+            the `~astropy.time.Time` initializer, so it can be anything that
+            `~astropy.time.Time` will accept (including a `~astropy.time.Time`
+            object)
+
+        target : coordinate object (i.e. `~astropy.coordinates.SkyCoord`, `~astroplan.FixedTarget`)
+            Target celestial object
+
+        horizon : `~astropy.units.Quantity` (optional), default = zero degrees
+            Degrees above/below actual horizon to use
+            for calculating rise/set times (i.e.,
+            -6 deg horizon = civil twilight, etc.)
+
+        ret_altaz : bool (optional)
+            Also return the '~astropy.coordinates.AltAz' coordinate.
+        '''
+        if not isinstance(time, Time):
+            time = Time(time)
+
+        altaz = self.altaz(time, target)
+        observable = bool(altaz.alt > horizon)
```

Comment:
This will fail if you the `altaz` multiple targets or times -- why do you wrap it with `bool`? The result of a comparison operator will either be a boolean (if scalar) or a boolean array